includes/handler/class-follow.php

```diff
@@ -40,13 +41,18 @@ public static function handle_follow( $activity, $user_ids ) {
 			return;
 		}
 
+		// Check if the actor already follows the user.
+		$already_following = false;
+		$existing_actor    = Remote_Actors::get_by_uri( $activity['actor'] );
+		if ( ! \is_wp_error( $existing_actor ) ) {
+			$already_following = Followers::follows( $existing_actor->ID, $user_id );
+		}
+
 		// Save follower.
-		$remote_actor = Followers::add(
-			$user_id,
-			$activity['actor']
-		);
+		$remote_actor = Followers::add( $user_id, $activity['actor'] );
```

Comment:
With that check in place, we can skip the Follow process when `$already_following` is `true`!?

```suggestion
		if ( ! $already_following ) {
			$remote_actor = Followers::add( $user_id, $activity['actor'] );
		}
```

Reply:
Fixed in [f67c6f4](https://github.com/Automattic/wordpress-activitypub/pull/2452/commits/f67c6f49e49d4b253865e52d734b77bd147df09c).